pxr/usd/plugin/usdDraco/exportAttribute.h

```diff
@@ -161,43 +170,50 @@ void UsdDracoExportAttribute<ArrayT>::SetToMesh(draco::Mesh *dracoMesh) {
     }
 }
 
-template <class ArrayT>
-void UsdDracoExportAttribute<ArrayT>::GetFromRange(size_t size) {
+template <class T>
+void UsdDracoExportAttribute<T>::GetFromRange(size_t size) {
     _MakeRange(&_values, size);
 }
 
-template <class ArrayT>
-void UsdDracoExportAttribute<ArrayT>::_MakeRange(
-        VtIntArray *array, size_t size) {
+template <class T>
+template <class S>
+void UsdDracoExportAttribute<T>::_MakeRange(
+        VtArray<S> *array, size_t size) {
     (*array).resize(size);
     for (size_t i = 0; i < size; i++) {
-        (*array)[i] = i;
+        (*array)[i] = static_cast<S>(i);
     }
 }
 
-template <class ArrayT>
-inline void UsdDracoExportAttribute<ArrayT>::_SetAttributeValue(
+template <class T>
+inline void UsdDracoExportAttribute<T>::_SetAttributeValue(
         draco::AttributeValueIndex avi, size_t index) {
-    _pointAttribute->SetAttributeValue(avi, _values[index].data());
+    _SetAttributeValueSpecialized(avi, _values[index], std::is_integral<T>());
 }
 
-template <>
-inline void UsdDracoExportAttribute<VtIntArray>::_SetAttributeValue(
-        draco::AttributeValueIndex avi, size_t index) {
-    _pointAttribute->SetAttributeValue(avi, &_values[index]);
+template<class T>
```

Comment:
I think it may be better to put this into a one function and do the std::is_integral<T> there:


void UsdDracoExportAttribute<T>::_SetAttributeValueSpecialized(
        draco::AttributeValueIndex avi, const T &value) {
    if (std::is_integral<T>::value)
      _pointAttribute->SetAttributeValue(avi, &value);
   else
     _pointAttribute->SetAttributeValue(avi, value.data());
}